### PR TITLE
Adding inbucket module

### PR DESCRIPTION
**Pull request by jespino**
## What does this PR do?

This PR adds the inbucket module

## Why is it important?

An important part of testing application usually involves sending emails,
inbucket helps a lot in this task acting like an MTA that you can inspect later
to do assertions.

## How to test this PR

I included unit tests that verifies the correct behavior

### CODE REVIEW

File: modules/inbucket/inbucket.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+)
+
+// InbucketContainer represents the Inbucket container type used in the module
+type InbucketContainer struct {
+	testcontainers.Container
+}
+
+// Smtp returns the connection string for the smtp server, using the default
+// 2500 port, and obtaining the host and exposed port from the container.
+func (c *InbucketContainer) Smtp(ctx context.Context) (string, error) {
```
Comment (mdelapenya) [inline modules/inbucket/inbucket.go:19]: Wdyt about this name?
```suggestion
func (c *InbucketContainer) SmtpConnection(ctx context.Context) (string, error) {
```

File: modules/inbucket/inbucket.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		return "", err
+	}
+
+	return net.JoinHostPort(host, containerPort.Port()), nil
+}
+
+// WebInterface returns the connection string for the web interface server,
+// using the default 9000 port, and obtaining the host and exposed port from
+// the container.
+func (c *InbucketContainer) WebInterface(ctx context.Context) (string, error) {
```
Comment (mdelapenya) [inline modules/inbucket/inbucket.go:36]: In an automated test suite, what would be the use case for this URL? is this something you plan to use in dev-environment, where all the backend services are running?

Comment (mdelapenya) [reply modules/inbucket/inbucket.go:36]: Ok, I've read the tests now, and this URL is exposing REST APIs. 👌

File: modules/inbucket/inbucket_test.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+		t.Fatal(err)
+	}
+	if len(headers) != 0 {
+		t.Fatalf("expected 0 messages, got %d", len(headers))
+	}
+
+	msg := []byte("To: to@example.org\r\n" +
+		"Subject: testcontainers test!\r\n" +
+		"\r\n" +
+		"This is a testcontainer test.\r\n")
```
Comment (mdelapenya) [inline modules/inbucket/inbucket_test.go:57]: Proper naming is important 😄 
```suggestion
		"This is a Testcontainers test.\r\n")
```

**Review by mdelapenya — COMMENTED** [review]
I left a few comments, not blockers, regarding names.

Other than that, LGTM, thanks!

File: modules/inbucket/inbucket_test.go (hunk 3 above)
Comment (jespino) [reply modules/inbucket/inbucket_test.go]: Done

File: modules/inbucket/inbucket.go (hunk 1 above)
Comment (jespino) [reply modules/inbucket/inbucket.go]: Done

File: modules/inbucket/inbucket_test.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	restClient, err := client.New(webInterfaceUrl)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	headers, err := restClient.ListMailbox("to@example.org")
```
Comment (mdelapenya) [inline modules/inbucket/inbucket_test.go:46]: We could make the container provide a way to retrieve the restClient, so the webInterface is not needed 🤔

Not an strong opinion on that, you are most familiar with the technology, so please consider keeping this as is.

Comment (mdelapenya) [reply modules/inbucket/inbucket_test.go:46]: Mmmm forget about that. I've just realised that if the container retrieves a `github.com/inbucket/inbucket/pkg/rest/client` type, then we'll be adding that dependency to the consumers (at the moment it's only present at test time).

**Review by mdelapenya — APPROVED** [review]
LGTM, thanks again for your hard work contributing modules!

**Review by mdelapenya — APPROVED** [review]
Great job again! Thanks for this new module, I'm pretty sure the community will thank having a way to test email interactions! 👏👏👏 

LGTM!

**Pull request merged**